src/xpath.c

```diff
@@ -151,6 +151,16 @@ print_token(enum lyxp_token tok)
     }
 }
 
+#define DBG_BUFF_SIZE 8192
+
+void dbg_sprintf_append(char *buff, size_t buff_size, char *fmt, ...)
+{
+    va_list argptr;
+    va_start(argptr,fmt);
+    size_t offset = strlen(buff);
+    snprintf(buff+offset, buff_size-offset-1, fmt, argptr);
```

Comment:
Did you test this? I think you wanted to use `vsnprintf(3)` instead. Also, put spaces around `+` and `-`.